scipy/optimize/optimize.py

```diff
@@ -2102,7 +2105,10 @@ def _minimize_scalar_golden(func, brack=None, args=(),
     f1 = func(*((x1,) + args))
     f2 = func(*((x2,) + args))
     funcalls += 2
-    while (numpy.abs(x3 - x0) > tol * (numpy.abs(x1) + numpy.abs(x2))):
+    nit = 0
+    for i in xrange(maxiter):
+        if (numpy.abs(x3 - x0) <= tol * (numpy.abs(x1) + numpy.abs(x2))):
```

Comment:
Not sure that that is possible. There are two possible cases in each update step

```python
x1, x2 = x2, _gR * x2 + _gC * x3
```

In this case we replace x1 to x2 and replace x2 to the weighted average between x2 and x3 points (because _gR + _gC = 1). Which means that the x3 shift new x2 towards itself. And the same is true for the other case.

```python
x1, x2 = _gR * x1 + _gC * x0, x1
```